src/utils/protectedRoute.jsx

```diff
@@ -1,11 +1,18 @@
 import { Navigate } from "react-router-dom";
 import { useAuth } from "./useAuth";
 import PropTypes from "prop-types";
+import { Spin } from "antd";
 
 //This component will be used to protect routes that require you to be signed in.
 const ProtectedRoute = ({ component: Component }) => {
-  const { isLoggedIn } = useAuth();
+  const { isLoggedIn, loading } = useAuth();
 
+  //wait for confrimLogin to finish before rendering the component
+  if (loading) {
+    return <Spin size="large" />;
+  }
+
+  // 'replace' modifies the history stack so you can't return to the analysis page via the back button.
   return isLoggedIn ? <Component /> : <Navigate to="/login" replace />;
```

Comment:
you can actually remove replace and you won't return to the analysis page while not logged in.

But apparently its a good safeguard to keep here for future components and situations.

Reply:
Ok lets update that comment then with what you know now

Reply:
@ErikAmerico 
github is kind of weird and my above comment was posted by itself as part of my review below so I just wanted to point out here that that comment is in response to your comment here ^^^^^^^^^